src/api/app/models/unregistered_user.rb

```diff
@@ -6,34 +6,41 @@ class UnregisteredUser < User
   class ErrRegisterSave < APIException
   end
 
+  # Raises an exception if registration is disabled for a user
+  # Returns true if a user can register
   def self.can_register?
     # No registering if LDAP is on
     if CONFIG['ldap_mode'] == :on
       logger.debug 'Someone tried to register with "ldap_mode" turned on'
-      raise ErrRegisterSave.new 'Sorry, new users can only sign up via LDAP'
+      raise ErrRegisterSave, 'Sorry, new users can only sign up via LDAP'
     end
+
     # No registering if we use an authentification proxy
-    if CONFIG['proxy_auth_mode'] == :on or CONFIG['ichain_mode'] == :on
+    if CONFIG['proxy_auth_mode'] == :on || CONFIG['ichain_mode'] == :on
       logger.debug 'Someone tried to register with "proxy_auth_mode" turned on'
-      if CONFIG['proxy_auth_register_page'].blank?
-        raise ErrRegisterSave.new "Sorry, please sign up using the authentification proxy"
+      err_msg = if CONFIG['proxy_auth_register_page'].blank?
```

Comment:
I don't like too much using ``if`` statements to return values for an assignment to a variable. Instead of that I would prefer the ternary operator something like ``(condition) ? "string1" : "string2"``.